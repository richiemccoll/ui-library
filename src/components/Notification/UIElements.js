import styled from 'styled-components';

export const getBackgroundColor = type => {
  switch (type) {
    case "error":
      return "red";
    case "warning":
      return "#777";
    case "success":
      return "#00E676";
    default:
      return "#c5c5c5";
  }
};

export const IconContainer = styled.span`
  position: absolute;
  right: 0;
  margin-right: 16px;
`;

export const Container = styled.div`
  width: 90%;
  padding: 8px;
  height: 24px;
  position: relative;
  background-color: ${props => props.type};
  border-radius: 2px;
`;