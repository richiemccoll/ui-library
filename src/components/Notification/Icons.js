import React from 'react';
import PropTypes from 'prop-types';
import { IconContainer } from './UIElements';

const CloseIcon = ({ handleClose }) => {
  return (
    <IconContainer onClick={handleClose}>
      <svg
        id="i-close"
        viewBox="0 0 32 32"
        width="16"
        height="16"
        fill="none"
        stroke="currentcolor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      >
        <path d="M2 30 L30 2 M30 30 L2 2" />
      </svg>
    </IconContainer>
  );
};

export default CloseIcon;

CloseIcon.propTypes = {
  handleClose: PropTypes.func.isRequired
}