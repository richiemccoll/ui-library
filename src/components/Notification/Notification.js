import React from "react";
import CloseIcon from './Icons';
import { getBackgroundColor, Container } from './UIElements';

const Notification = ({ type, message, closeNotification }) => {
  return (
    <Container type={getBackgroundColor(type)} role="alert">
      {message}
      <CloseIcon handleClose={closeNotification} />
    </Container>
  );
};

export default Notification;
