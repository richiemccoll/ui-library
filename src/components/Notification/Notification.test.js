import React from "react";
import renderer from "react-test-renderer";
import Notification from "./Notification";

describe("Notification Component", () => {
  it("should render correctly", () => {
    const component = renderer.create(
      <Notification type="message" message="This is a normal message variant" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
