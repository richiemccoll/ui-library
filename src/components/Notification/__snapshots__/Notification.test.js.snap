// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component should render correctly 1`] = `
<div
  className="sc-bwzfXH jNMMzs"
  role="alert"
  type="#c5c5c5"
>
  This is a normal message variant
  <span
    className="sc-bdVaJa fNAerP"
    onClick={undefined}
  >
    <svg
      fill="none"
      height="16"
      id="i-close"
      stroke="currentcolor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 32 32"
      width="16"
    >
      <path
        d="M2 30 L30 2 M30 30 L2 2"
      />
    </svg>
  </span>
</div>
`;
